src/store/slices/products/products.thunks.ts

```diff
@@ -3,12 +3,20 @@ import axiosInstance from '@/services/axiosInstance';
 import {handleApiError} from '@/services/handleApiError';
 import {createAsyncThunk} from '@reduxjs/toolkit';
 import {Product} from './products.model';
+import {Filters} from '../filters/filters.model';
 
 export const getAllProducts = createAsyncThunk(
   'products/getAllProducts',
-  async (_, {rejectWithValue}): Promise<Product[] | any> => {
+  async (filters: Filters | {} = {}, {rejectWithValue}): Promise<Product[] | any> => {
     try {
-      return await axiosInstance.get(getAllProductsUrl);
+      let params: Record<string, any> = {};
+      for (const [key, value] of Object.entries(filters)) {
+        if (value) {
+          params[key] = value;
+        }
+      }
```

Comment:
For of for array is not the best. You can use for each instead or don`t convert it to the array and work with the object.

Reply:
Changed to: 
```
let params: Record<string, any> = {};
      Object.entries(filters).forEach(([key, value]) => {
        if (value) {
          params[key] = value;
        }
      });
```